ping.go

```diff
@@ -370,15 +372,29 @@ func (p *Pinger) Run() error {
 			}
 		}
 		if p.Count > 0 && p.PacketsRecv >= p.Count {
-			close(p.done)
+			_ = p.Stop()
 			wg.Wait()
 			return nil
 		}
 	}
 }
 
-func (p *Pinger) Stop() {
+func (p *Pinger) Stop() error {
```

Comment:
if we do end up returning ErrTownDown when Stop() has already been called, we should document that behavior in a comment on this function